qcodes/dataset/data_set.py

```diff
@@ -1247,15 +1255,25 @@ def _load_to_xarray_dataset(self, data: ParameterData) -> "xr.Dataset":
                 xrdataset.coords[str(dim)].attrs.update(paramspec_dict.items())
 
         xrdataset.attrs.update({
```

Comment:
note that this PR only adds attributes to xarray DATASET, but we also have the xarray dataarrays, `_load_to_xarray_dataarray_dict`, shall we add some attributes to them as well? i'm thinking at least name,exp_name,sample_name,guid,run_description (or even fancier, run_description of the contents of the dataarray only ;) )